app/models/concerns/info_request/title_validation.rb

```diff
@@ -0,0 +1,66 @@
+# Everything to do with validating an InfoRequest title
+module InfoRequest::TitleValidation
+  extend ActiveSupport::Concern
+  include InfoRequest::DraftTitleValidation
+
+  included do
+    validates_presence_of :title,
+                          message: N_('Please enter a summary of your request')
+
+    validates_format_of :title,
+      with: /\A.*[[:alpha:]]+.*\z/,
+      message: N_('Please write a summary with some text in it'),
+      unless: proc { |record| record.title.blank? }
+
+    validates :title, length: {
+      minimum: 3,
+      message: _('Summary is too short. Please be a little more descriptive ' \
+                 'about the information you are asking for.'),
+      unless: proc { |record| record.title.blank? },
+      on: :create
+    }
+
+    # only check on create, so existing models with mixed case are allowed
+    validate :title_formatting, on: :create
+  end
+
+  private
+
+  def title_formatting
+    return unless title
+
+    if poorly_formed_title?
+      errors.add(:title, _('Please write the summary using a mixture of ' \
+                           'capital and lower case letters. This makes it ' \
+                           'easier for others to read.'))
+    end
+
+    if generic_foi_title?
```

Comment:
-1 to the rubocop warning here.

Reply:
Well, actually maybe _this_ is nicer:

```diff
diff --git a/app/models/concerns/info_request/title_validation.rb b/app/models/concerns/info_request/title_validation.rb
index beb389268..ea4e3b827 100644
--- a/app/models/concerns/info_request/title_validation.rb
+++ b/app/models/concerns/info_request/title_validation.rb
@@ -28,18 +28,8 @@ module InfoRequest::TitleValidation
 
   def title_formatting
     return unless title
-
-    if poorly_formed_title?
-      errors.add(:title, _('Please write the summary using a mixture of ' \
-                           'capital and lower case letters. This makes it ' \
-                           'easier for others to read.'))
-    end
-
-    if generic_foi_title?
-      errors.add(:title, _('Please describe more what the request is about ' \
-                           'in the subject. There is no need to say it is an ' \
-                           'FOI request, we add that on anyway.'))
-    end
+    errors.add(:title, poorly_formed_title_msg) if poorly_formed_title?
+    errors.add(:title, generic_foi_title_msg) if generic_foi_title?
   end
 
   def title_is_acronym(max_length)
@@ -60,7 +50,19 @@ def well_formed_title?
       title_is_acronym(6)
   end
 
+  def poorly_formed_title_msg
+    _('Please write the summary using a mixture of ' \
+      'capital and lower case letters. This makes it ' \
+      'easier for others to read.')
+  end
+
   def generic_foi_title?
     title =~ /^(FOI|Freedom of Information)\s*requests?$/i
   end
+
+  def generic_foi_title_msg
+    _('Please describe more what the request is about ' \
+      'in the subject. There is no need to say it is an ' \
+      'FOI request, we add that on anyway.')
+  end
 end
```

Could alternatively make the messages constants instead of methods?

Reply:
I've pushed this version (6c731b0). I think methods _feel_ nicer in this context.